### PR TITLE
Add flag to enable latency metrics

### PR DESCRIPTION
**Pull request by sonjaer**
# TL;DR
We would like latency metrics on the outgoing RPCs. I found in the doc that they can be enabled by doing [this](https://github.com/grpc-ecosystem/go-grpc-prometheus#histograms). Histogram metrics / metrics with high cardinality can be expensive on Prometheus servers, so I added it behind a flag (see [here](https://github.com/grpc-ecosystem/go-grpc-prometheus/blob/master/client.go#L32)).

https://github.com/flyteorg/flyte/issues/2489

## Type
 - [ ] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
_Remove the '*fixes*' keyword if there will be multiple PRs to fix the linked issue_

fixes https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

**Review by katrogan — DISMISSED** [review]

**Comment by hamersaw** [comment]
Quick question on this. Is the goal to use the histograms for SLO's or just generalized observation? The [Prometheus best practices on histograms](https://prometheus.io/docs/practices/histograms/) is a great read, it has helped me understand this a bit better, including the performance implications. My brief takeaways were: if you want an SLO on quantile values use summaries (rather than histograms) and if you want to use histograms it is very important to properly configure the bucket sizes, using default values may result in all observations in a single bucket and increasing the observation cardinality (defining > 10 buckets) has significant impact on performance.

**Comment by sonjaer** [comment]
> generalized observation



> Quick question on this. Is the goal to use the histograms for SLO's or just generalized observation? The [Prometheus best practices on histograms](https://prometheus.io/docs/practices/histograms/) is a great read, it has helped me understand this a bit better, including the performance implications. My brief takeaways were: if you want an SLO on quantile values use summaries (rather than histograms) and if you want to use histograms it is very important to properly configure the bucket sizes, using default values may result in all observations in a single bucket and increasing the observation cardinality (defining > 10 buckets) has significant impact on performance.

It is for generalized observation

**Comment by hamersaw** [comment]
merged in [this PR](https://github.com/flyteorg/flytepropeller/pull/460)
